src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -2063,6 +2063,55 @@ closeGraphContext(unsigned int graphId)
   return ret ? -errno : ret;
 }
 
+int
+shim::
+open_graph_context(const zynqaie::hwctx_object* hwctx, const uuid_t xclbinId, unsigned int graph_id, xrt::graph::access_mode am)
+{
+  int ret = 0;
+  if (!hw_context_enable){
+    // for legacy flow
+    ret = openGraphContext(xclbinId, graph_id, am);
+    if (ret)
+      throw xrt_core::error("Failed to open graph context");
+    return ret;
+  }
+  else {
+    // this is for multi slot case
+    auto shared = (hwctx->get_mode() != xrt::hw_context::access_mode::exclusive);
+    unsigned int flags = shared ? ZOCL_CTX_SHARED : ZOCL_CTX_EXCLUSIVE;
+    drm_zocl_open_graph_ctx graph_ctx = {};
+    graph_ctx.hw_context = hwctx->get_slotidx();
+    graph_ctx.flags = flags;
+    graph_ctx.graph_id = graph_id;
+    ret = ioctl(mKernelFD, DRM_IOCTL_ZOCL_OPEN_GRAPH_CTX, &graph_ctx);
+    if (ret)
+      throw xrt_core::error("Failed to open graph context hwctx shim");
+    return ret;
+  }
+}
```

Comment:
The return value communicates nothing.  Please make API void and throw on error.

```suggestion
void
shim::
open_graph_context(const zynqaie::hwctx_object* hwctx, const uuid_t xclbinId, unsigned int graph_id, xrt::graph::access_mode am)
{
  if (!hw_context_enable) {
    // for legacy flow
    if (auto ret = openGraphContext(xclbinId, graph_id, am))
      throw xrt_core::error(ret, "Failed to open graph context");
      
    return;
  }
  // this is for multi slot case
  drm_zocl_open_graph_ctx graph_ctx = {};
  graph_ctx.hw_context = hwctx->get_slotidx();
  graph_ctx.flags =  (hwctx->get_mode() != xrt::hw_context::access_mode::exclusive) ? ZOCL_CTX_SHARED : ZOCL_CTX_EXCLUSIVE
  graph_ctx.graph_id = graph_id;
  if (auto ret = ioctl(mKernelFD, DRM_IOCTL_ZOCL_OPEN_GRAPH_CTX, &graph_ctx)_
      throw xrt_core::error(ret, "Failed to open graph context hwctx shim");
}
```

Reply:
Hi @stsoe thank you for the review I've updated with latest push.